lib/ansible/modules/cloud/vmware/vmware_guest_cross_vc_clone.py

```diff
@@ -0,0 +1,292 @@
+                                         wait_for_task, TaskError)
+from ansible.module_utils._text import to_native
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+
+class CloneManager(PyVmomi):
+    def __init__(self, module):
+        super(CloneManager, self).__init__(module)
+        self.config_spec = vim.vm.ConfigSpec()
+        self.clone_spec = vim.vm.CloneSpec()
+        self.relocate_spec = vim.vm.RelocateSpec()
+        self.service_locator = vim.ServiceLocator()
+        self.destination_vcenter = self.params['destination_vcenter']
+        self.destination_vcenter_username = self.params['destination_vcenter_username']
+        self.destination_vcenter_password = self.params['destination_vcenter_password']
+
+    def get_new_vm_info(self, vm):
+        # to check if vm has been cloned in the destination vc
+        # query for the vm in destination vc
+        # get the host and datastore info
+        # get the power status of the newly cloned vm
+        info = {}
+        vm_obj = find_vm_by_name(content=self.destination_content, vm_name=vm)
+        if vm_obj is None:
+            module.fail_json(msg="Newly cloned VM is not found in the destination VCenter")
+        else:
+            vm_facts = gather_vm_facts(self.destination_content, vm_obj)
+            info['vm_name'] = vm
+            info['vcenter'] = self.destination_vcenter
+            info['host'] = vm_facts['hw_esxi_host']
+            info['datastore'] = vm_facts['hw_datastores']
+            info['vm_folder'] = vm_facts['hw_folder']
+            info['power_on'] = vm_facts['hw_power_status']
+        return info
+
+    def clone(self):
+        # clone the vm/template on destination VC
+        vm_folder = find_folder_by_name(content=self.destination_content, folder_name=self.params['destination_vm_folder'])
+        vm_name = self.params['destination_vm_name']
+        task = self.vm_obj.Clone(folder=vm_folder, name=vm_name, spec=self.clone_spec)
+        wait_for_task(task)
+        if task.info.state == 'error':
+            result = {'changed': False, 'failed': True, 'msg': task.info.error.msg}
+        else:
+            vm_info = self.get_new_vm_info(vm_name)
+            result = {'changed': True, 'failed': False, 'vm_info': vm_info}
+        return result
+
+    def sanitize_params(self):
+        '''
+        this method is used to verify user provided parameters
+        '''
+        self.vm_obj = find_vm_by_name(content=self.content, vm_name=self.params['name'])
+        if self.vm_obj is None:
+            vm_id = self.vm_uuid or self.vm_name or self.moid
+            self.module.fail_json(msg="Failed to find the VM/template with %s" % vm_id)
+
+        # connect to destination VC
+        self.destination_content = connect_to_api(
+            self.module,
+            hostname=self.destination_vcenter,
+            username=self.destination_vcenter_username,
+            password=self.destination_vcenter_password)
```

Comment:
pass `port` and `validate_certs` as well. Since these details can be different for destination datacenter.